### PR TITLE
Added sha256 support for generating file hash

### PR DESCRIPTION
**Pull request by nishantnahata**

### CODE REVIEW

File: tests/unit_tests.cpp [hunk 1]
```diff
@@ -129,3 +130,6 @@ void dfr::file_hash_test::md5_hash_test() {
     CPPUNIT_ASSERT_EQUAL(std::string(""), dfr::hash::MD5("non_existent_dir"));
 }
 
+void dfr::file_hash_test::sha256_hash_test() {
+    CPPUNIT_ASSERT_EQUAL(std::string(""), dfr::hash::SHA256("non_existent_dir"));
```
Comment (vishal-wadhwa) [inline tests/unit_tests.cpp:135]: Please add two more tests similar to the ones in `md5_hash_test`.

File: tests/unit_tests.cpp [hunk 2]
```diff
@@ -5,6 +5,7 @@
 #include <unordered_map>
 #include <string>
 #include <fstream>
+#include<iostream>
```
Comment (vishal-wadhwa) [inline tests/unit_tests.cpp]: Uh! I don't think `iostream` is needed in this file. Please remove. :)

File: src/file_handler.cpp [hunk 3]
```diff
@@ -6,12 +6,13 @@
 #include <functional>
 #include <iomanip>
 #include <queue>
+#include<math.h>
```
Comment (vishal-wadhwa) [inline src/file_handler.cpp]: `#include <cmath>` would be a better choice. Read [this](https://stackoverflow.com/questions/15656434/include-cmath-vs-include-math-h-in-a-c-program).

**Review by vishal-wadhwa — CHANGES_REQUESTED** [review]
Great work, man!

There are some minor changes I've suggested please make them and commit.

Thanks :)

File: tests/unit_tests.cpp (hunk 2 above)
Comment (nishantnahata) [reply tests/unit_tests.cpp]: ohh! yes, I added it for some manual testing, i will remove it

**Review by vishal-wadhwa — APPROVED** [review]

**Comment by vishal-wadhwa** [comment]
Fixes issue #2

**Pull request merged**